C/services/south/ingest.cpp

```diff
@@ -739,23 +759,43 @@ void Ingest::processQueue()
 
 				string lastAsset;
 				int *lastStat = NULL;
-				std::map <std::string , std::set<std::string> >assetDatapointMap;
+
+				std::map <std::string, dpCountObj> assetDatapointMap;
 
 				for (vector<Reading *>::iterator it = m_data->begin(); it != m_data->end(); ++it)
 				{
 		               	        Reading *reading = *it;
 					string	assetName = reading->getAssetName();
-
 					const std::vector<Datapoint *> dpVec = reading->getReadingData();
+					std::string temp;
+                                        std::set<std::string> tempSet;
+                                        // first sort the individual datapoints
+                                        // e.g. dp2, dp3, dp1 push them in a set,to make them
+                                        // dp1,dp2,dp3
 
-					for ( auto dp : dpVec)
-					{
-						std::string temp;
-						temp.append("\"");
-						temp.append(dp->getName());
-						temp.append("\"");
-						assetDatapointMap[assetName].insert(temp);
-					}
+                                        for ( auto dp : dpVec)
+                                        {
+						temp.clear();
+                                                temp.append("\"");
```

Comment:
do we need enclosing quotes?

Reply:
as per requirement on jIRA

Reply:
As per JIRA, "datapoints" is an array of dp name strings, and in JSON, strings are double quoted, there is no such requirement in C++ code, only when conversion to JSON happens. Can you show me sample JSON from ur testing?

Reply:
This might be a choice issue , whether to keep it here or adding in management-client.cpp